### ISSUE TITLE
Poor metrics when training with GTR-base on NQ dataset

### ISSUE THREAD

**Issue by ChrisWang233** [jxmorris12/vec2text#99]
Thank you for sharing your work. I was attempting to train the embedding inversion model, I downloaded the code and run the following command right the way:

python run.py \
  --per_device_train_batch_size 128 \
  --per_device_eval_batch_size 128 \
  --max_seq_length 128 \
  --model_name_or_path t5-base \
  --dataset_name nq \
  --embedder_model_name gtr_base \
  --num_repeat_tokens 16 \
  --embedder_no_grad True \
  --num_train_epochs 100 \
  --max_eval_samples 500 \
  --eval_steps 20000 \
  --warmup_steps 10000 \
  --bf16=1 \
  --use_wandb=1 \
  --use_frozen_embeddings_as_input True \
  --experiment inversion \
  --lr_scheduler_type constant_with_warmup \
  --exp_group_name oct-gtr \
  --learning_rate 0.001 \
  --output_dir ./saves/gtr-1 \
  --save_steps 2000

However, after 5 epochs, the metrics remain significantly low. The training loss is stuck between 2.0 and 3.0, and the BLEU score is only around 1-2, with no visible upward trend. I feel something is wrong with the current setup as it's so much worse than your pretrained model [gtr__nq__32]. Could you advise some common pitfalls when training this inversion model? Additionally, should I expect meaningful convergence only toward the end of the 100-epoch schedule?

**Comment by jxmorris12** [comment]
It's hard to diagnose without seeing the loss and evaluation performance. Is the loss going down? And if so, how much training data is there?

**Comment by ChrisWang233** [comment]
> It's hard to diagnose without seeing the loss and evaluation performance. Is the loss going down? And if so, how much training data is there?

The loss is kind of sticking around 2.7, I used full training data of nq dataset. Validation result on in-domain set is ok, but on other val sets remain poor.


{'loss': 2.7218, 'grad_norm': 0.2285870611667633, 'learning_rate': 0.001, 'epoch': 4.714198264376508}
{'loss': 2.7186, 'grad_norm': 1.1235147714614868, 'learning_rate': 0.001, 'epoch': 4.743149328436139}
{'loss': 2.7218, 'grad_norm': 0.21429918706417084, 'learning_rate': 0.001, 'epoch': 4.7986388678837635}
{'loss': 2.7178, 'grad_norm': 0.4337350130081177, 'learning_rate': 0.001, 'epoch': 4.957869720211732}

{'eval_ag_news_loss': 3.511345624923706, 'eval_ag_news_accuracy': 0.1831875, 'eval_ag_news_pred_num_tokens': 126.15625, 'eval_ag_news_true_num_tokens': 65.984375, 'eval_ag_news_token_set_precision': 0.2753871743029181, 'eval_ag_news_token_set_recall': 0.35010573182492044, 'eval_ag_news_token_set_f1': 0.2935955749934092, 'eval_ag_news_token_set_f1_sem': 0.004113752204760624, 'eval_ag_news_n_ngrams_match_1': 13.616, 'eval_ag_news_n_ngrams_match_2': 2.626, 'eval_ag_news_n_ngrams_match_3': 0.734, 'eval_ag_news_num_true_words': 45.122, 'eval_ag_news_num_pred_words': 94.234, 'eval_ag_news_bleu_score': 2.1956976745455266, 'eval_ag_news_bleu_score_sem': 0.07583834539845606, 'eval_ag_news_rouge_score': 0.20463132957486535, 'eval_ag_news_exact_match': 0.0, 'eval_ag_news_exact_match_sem': 0.0, 'eval_ag_news_emb_cos_sim': 0.7381443381309509, 'eval_ag_news_emb_cos_sim_sem': 0.006569574499959882, 'eval_ag_news_emb_top1_equal': 0.1354166716337204, 'eval_ag_news_emb_top1_equal_sem': 0.0247584133890085, 'eval_ag_news_perplexity': 33.493306902814176, 'eval_ag_news_runtime': 18.7781, 'eval_ag_news_samples_per_second': 26.627, 'eval_ag_news_steps_per_second': 0.16, 'epoch': 5.101040653919556}

{'eval_anthropic_toxic_prompts_loss': 2.9904775619506836, 'eval_anthropic_toxic_prompts_accuracy': 0.061359375, 'eval_anthropic_toxic_prompts_pred_num_tokens': 122.28125, 'eval_anthropic_toxic_prompts_true_num_tokens': 18.109375, 'eval_anthropic_toxic_prompts_token_set_precision': 0.33353786338369, 'eval_anthropic_toxic_prompts_token_set_recall': 0.32027928948760015, 'eval_anthropic_toxic_prompts_token_set_f1': 0.30005200495492357, 'eval_anthropic_toxic_prompts_token_set_f1_sem': 0.006697017006530165, 'eval_anthropic_toxic_prompts_n_ngrams_match_1': 4.776, 'eval_anthropic_toxic_prompts_n_ngrams_match_2': 1.326, 'eval_anthropic_toxic_prompts_n_ngrams_match_3': 0.486, 'eval_anthropic_toxic_prompts_num_true_words': 14.816, 'eval_anthropic_toxic_prompts_num_pred_words': 95.3, 'eval_anthropic_toxic_prompts_bleu_score': 1.190605100072488, 'eval_anthropic_toxic_prompts_bleu_score_sem': 0.046306165290941954, 'eval_anthropic_toxic_prompts_rouge_score': 0.0945095074371759, 'eval_anthropic_toxic_prompts_exact_match': 0.0, 'eval_anthropic_toxic_prompts_exact_match_sem': 0.0, 'eval_anthropic_toxic_prompts_emb_cos_sim': 0.610969066619873, 'eval_anthropic_toxic_prompts_emb_cos_sim_sem': 0.007797145317450485, 'eval_anthropic_toxic_prompts_emb_top1_equal': 0.02604166604578495, 'eval_anthropic_toxic_prompts_emb_top1_equal_sem': 0.011523592621071275, 'eval_anthropic_toxic_prompts_perplexity': 19.895181404865326, 'eval_anthropic_toxic_prompts_runtime': 16.2268, 'eval_anthropic_toxic_prompts_samples_per_second': 30.813, 'eval_anthropic_toxic_prompts_steps_per_second': 0.185, 'epoch': 5.101040653919556}

{'eval_nq_loss': 2.4086575508117676, 'eval_nq_accuracy': 0.496828125, 'eval_nq_pred_num_tokens': 124.546875, 'eval_nq_true_num_tokens': 127.1484375, 'eval_nq_token_set_precision': 0.2928897600517297, 'eval_nq_token_set_recall': 0.5012634313686376, 'eval_nq_token_set_f1': 0.362491012438716, 'eval_nq_token_set_f1_sem': 0.003935136748126754, 'eval_nq_n_ngrams_match_1': 39.408, 'eval_nq_n_ngrams_match_2': 12.18, 'eval_nq_n_ngrams_match_3': 4.896, 'eval_nq_num_true_words': 100.204, 'eval_nq_num_pred_words': 98.356, 'eval_nq_bleu_score': 7.76572751076923, 'eval_nq_bleu_score_sem': 0.32284715932696656, 'eval_nq_rouge_score': 0.3676958489604114, 'eval_nq_exact_match': 0.0, 'eval_nq_exact_match_sem': 0.0, 'eval_nq_emb_cos_sim': 0.806045413017273, 'eval_nq_emb_cos_sim_sem': 0.006420148929006936, 'eval_nq_emb_top1_equal': 0.65625, 'eval_nq_emb_top1_equal_sem': 0.03436681882528361, 'eval_nq_perplexity': 11.11902439612076, 'eval_nq_runtime': 14.7751, 'eval_nq_samples_per_second': 33.841, 'eval_nq_steps_per_second': 0.203, 'epoch': 5.101040653919556}
